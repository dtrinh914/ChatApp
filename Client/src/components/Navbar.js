import React from 'react';
import './Navbar.css'
import axios from 'axios'

function Navbar({username, history, setUserData}){
    const handleClick = () => {
        axios.get('/api/users/logout', {withCredentials:true})
        .then(res => {
            if(res.data.loggedIn === false){
                setUserData(res.data);
                history.push('/');
            }
        })
        .catch((err) => console.log(err));
    }
    return(
        <nav className='Navbar'>
            <h1>{username}</h1>
            <button onClick={handleClick}>Log Out</button>
        </nav>
    )
}
export default Navbar;