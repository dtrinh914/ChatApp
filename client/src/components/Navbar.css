.Navbar{
    height: 5vh;
    background: black;
    color: white;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 15px;
}

.Navbar button{
    border: none;
    background: none;
    color: white;
    cursor: pointer;
    height: 100%;
}