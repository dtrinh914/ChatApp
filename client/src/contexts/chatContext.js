import React, {createContext, useReducer} from 'react';
import chatReducer from '../reducers/chatReducer';

const initialState = {user:"", groups:"", selected:""}

export const ChatContext = createContext();

export function ChatProvider(props){
    const [chatData,chatDispatch] = useReducer(chatReducer, initialState);
    return(
        <ChatContext.Provider value={{chatData, chatDispatch}}>
            {props.children}
        </ChatContext.Provider>
    )
}