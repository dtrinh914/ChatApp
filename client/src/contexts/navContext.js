import React, {createContext, useReducer} from 'react';
import navReducer from '../reducers/navReducer';

const initialState = {
    rightNav:{
        root:false,
        addMem:false
    }
}

export const NavContext = createContext();

export function NavProvider(props){
    const [navData , navDispatch] = useReducer(navReducer, initialState);
    return(
        <NavContext.Provider value={{navData, navDispatch}}>
            {props.children}
        </NavContext.Provider>
    )
}