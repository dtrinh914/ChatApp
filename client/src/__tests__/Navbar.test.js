import React from 'react';
import Navbar from '../components/Navbar';
import { ChatProvider } from '../contexts/chatContext';
import { NavProvider} from '../contexts/navContext';
import {render} from '@testing-library/react';

test('should match snapshot', ()=>{
    const {container} = render(<ChatProvider>
                             <NavProvider>
                                <Navbar />       
                             </NavProvider>
                             </ChatProvider>);
    expect(container).toMatchSnapshot();
})