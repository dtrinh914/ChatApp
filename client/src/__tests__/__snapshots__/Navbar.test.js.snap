// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic makeStyles-nav-1 MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-tool-2 MuiToolbar-gutters"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
        data-testid="nav-group-name"
      />
    </div>
  </header>
</div>
`;
